acados/ocp_nlp/ocp_nlp_cost_ls.c

```diff
@@ -607,22 +661,51 @@ void ocp_nlp_cost_ls_update_qp_matrices(void *config_, void *dims_,
 
     int nx = dims->nx;
     int nu = dims->nu;
+    int nz = dims->nz;
     int ny = dims->ny;
     int ns = dims->ns;
 
     // initialize hessian of lagrangian with hessian of cost
     blasfeo_dgecp(nu + nx, nu + nx, &memory->hess, 0, 0, memory->RSQrq, 0, 0);
 
+    // copy Cyt into Cyt_tilde
+    blasfeo_dgecp(nu + nx, ny, &model->Cyt, 0, 0, &work->Cyt_tilde, 0, 0);
+    // copy y_ref into y_ref_tilde
+    blasfeo_dveccp(ny, &model->y_ref, 0, &work->y_ref_tilde, 0);
+    if (nz > 0) { // eliminate algebraic variables and update Cyt and y_ref
+        // swap dzdu and dzdx
+        blasfeo_dgecp(nx, ny, memory->dzdxu_tran, 0, 0,  &work->dzdux_tran, nu, 0);
+        blasfeo_dgecp(nu, ny, memory->dzdxu_tran, nx, 0, &work->dzdux_tran, 0, 0);
+        // update Cyt: Cyt_tilde = Cyt + dzdux_tran*Vz^T
+        blasfeo_dgemm_nt(nu + nx, ny, nz, 1.0, &work->dzdux_tran, 0, 0,
+                &model->Vz, 0, 0, 1.0, &work->Cyt_tilde, 0, 0, &work->Cyt_tilde, 0, 0);
+
+        // update y_ref: y_ref_tilde = y_ref + Vz*x + Vz*u - Vz*z
+        blasfeo_dveccp(nz, &memory->z, 0, &work->tmp_nz, 0);
```

Comment:
No need to copy: this can be taken care from the 3rd argument of the next `blasfeo_dgemv_t`

Reply:
addressed here f4181ec